novawallet/Modules/PayCard/Mercuryo/Hooks/MercuryoCardHookFactory+ResponseIntercept.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+
+extension MercuryoCardHookFactory {
+    func createResponseInterceptingHook(
+        using params: MercuryoCardParams,
+        for delegate: PayCardHookDelegate
+    ) -> PayCardHook {
+        let cardsAction = MercuryoMessageName.onCardsResponse.rawValue
+        let topUpAction = MercuryoMessageName.onCardTopup.rawValue
+
+        let scriptSource = """
+        let originalXhrOpen = XMLHttpRequest.prototype.open;
+
+        XMLHttpRequest.prototype.open = function(method, url) {
+            if (url === '\(MercuryoCardApi.cardsEndpoint)') {
+                this.addEventListener('load', function() {
+                    window.webkit.messageHandlers.\(cardsAction).postMessage(this.responseText);
+                });
+            }
+
+            if (url.includes('\(MercuryoCardApi.topUpEndpoint)')) {
```

Comment:
We check for inclusion here because the endpoint is just a part of url with params. In the first case we have full endpoint that have no params so we can check for equality.